### PR TITLE
CLIP-1520: Make ATL_FORCE_CFG_UPDATE configurable in values.yaml

### PR DESCRIPTION
**Pull request by bianchi2**
This PR introduces a new env to values.yaml to add `ATL_FORCE_CFG_UPDATE` to container env if `<product-name>.forceConfigUpdate` is set to true.

The following files are affected:
* Jira: `dbconfig.xml`
* Confluence: `confluence.cfg.xml`
* Bamboo: `bamboo.cfg.xml`

The default value is false, i.e. the above config files are not overwritten, i.e. nothing is changed in terms of previous defaults. This can be discussed though

## Checklist
- [x] I have added unit tests
- [x] I have applied the change to all applicable products
- [ ] I have added the change description to the `changelog.md` and `Chart.yaml` files
- [ ] (Atlassian only) I have run the E2E test (if applicable)
- [ ] (Atlassian only) Internal Bamboo CI is passing

### CODE REVIEW

**Review by nanux — CHANGES_REQUESTED** [review]

File: docs/docs/userguide/CONFIGURATION.md [hunk 1]
```diff
@@ -259,6 +259,33 @@ By default, the Helm charts will not configure the products for Data Center clus
...
+* Jira: `dbconfig.xml`
+* Confluence: `confluence.cfg.xml`
+* Bamboo: `bamboo.cfg.xml`
+
+To force update of the configuration files when pods restart, set `<product_name.forceConfigUpdate>` to true.
+You can do it by passing an argument to helm install/update command:
+```
+--set jira.forceConfigUpdate=true
+```
+or set it in values.yaml:
```
[the remark this answers is not in the document]
Comment (bianchi2) [reply docs/docs/userguide/CONFIGURATION.md]: Fixed

File: src/main/charts/bamboo/values.yaml [hunk 2]
```diff
@@ -632,6 +632,13 @@ bamboo:
       #    cpu: "2"
       #    memory: "2G"
 
+  # -- The Docker entrypoint.py generates application configuration on
+  # first start; not all of these files are regenerated on subsequent starts.
+  # By default, bamboo.cfg.xml is generated only once. Set configOverwrite to true
```
[the remark this answers is not in the document]
Comment (bianchi2) [reply src/main/charts/bamboo/values.yaml]: Fixed

File: src/main/charts/bamboo/values.yaml [hunk 3]
```diff
@@ -632,6 +632,13 @@ bamboo:
...
+  # to change this behavior. ATL_FORCE_CFG_UPDATE=true will be added to container env.
```
[the remark this answers is not in the document]
Comment (bianchi2) [reply src/main/charts/bamboo/values.yaml]: Removed

File: src/main/charts/jira/values.yaml [hunk 4]
```diff
@@ -546,6 +546,12 @@ jira:
       #    cpu: "2"
       #    memory: "2G"
 
+  # -- The Docker entrypoint.py generates application configuration on
+  # first start; not all of these files are regenerated on subsequent starts.
+  # By default, dbconfig.xml is generated only once. Set configOverwrite to true
```
[the remark this answers is not in the document]
Comment (bianchi2) [reply src/main/charts/jira/values.yaml]: Fixed

File: src/test/java/test/ForceCfgUpdateEnvVarTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import test.helm.Helm;
+import test.model.Product;
+
+import java.util.Map;
+
+
+/**
+ * Tests the various permutations of the "<product>.license" value structure in the Helm charts
+ */
+class ForceCfgUpdateEnvVarTest {
```
[the remark this answers is not in the document]
Comment (bianchi2) [reply src/test/java/test/ForceCfgUpdateEnvVarTest.java]: Renamed

File: src/test/java/test/ForceCfgUpdateEnvVarTest.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    private Helm helm;
+
+    @BeforeEach
+    void initHelm(TestInfo testInfo) {
+        helm = new Helm(testInfo);
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = Product.class, names = "jira")
```
[the remark this answers is not in the document]
Comment (bianchi2) [reply src/test/java/test/ForceCfgUpdateEnvVarTest.java]: I thought it's the same templating logic so one product is enough. Added the other two.

**Review by tarka — APPROVED** [review]
As discussed elsewhere, this isn't guaranteed to stick with Confluence, as it doesn't necessarily honour all values in a provided `confluence.cfg.xml`. However it will help in the case that a bad config needs to be reset, which is what I believe the original issue was here.

**Review by nanux — APPROVED** [review]

**Pull request merged**